pkg/utils/kubernetes/reduce/filters.go

```diff
@@ -115,14 +144,41 @@ func filterClusterRoleBindings(clusterRoleBindings []rbacv1.ClusterRoleBinding)
 		return []rbacv1.ClusterRoleBinding{}
 	}
 
-	toFilter := 0
+	oldestWithManagedByLabels := -1
+	oldestLegacy := -1
 	for i, clusterRoleBinding := range clusterRoleBindings {
-		if clusterRoleBinding.CreationTimestamp.Before(&clusterRoleBindings[toFilter].CreationTimestamp) {
-			toFilter = i
+		labels := clusterRoleBinding.GetLabels()
+
+		_, okManagedBy := labels[consts.GatewayOperatorManagedByLabel]
+		_, okManagedByNs := labels[consts.GatewayOperatorManagedByNamespaceLabel]
+		_, okManagedByName := labels[consts.GatewayOperatorManagedByNameLabel]
+		if okManagedBy && okManagedByNs && okManagedByName {
+			if oldestWithManagedByLabels == -1 {
+				oldestWithManagedByLabels = i
+				continue
+			}
+
+			if clusterRoleBinding.CreationTimestamp.Before(&clusterRoleBindings[oldestWithManagedByLabels].CreationTimestamp) {
+				oldestWithManagedByLabels = i
+			}
+			continue
 		}
+
+		if oldestLegacy == -1 {
+			oldestLegacy = i
+			continue
+		}
+
+		if clusterRoleBinding.CreationTimestamp.Before(&clusterRoleBindings[oldestLegacy].CreationTimestamp) {
+			oldestLegacy = i
+		}
+		continue
 	}
 
-	return append(clusterRoleBindings[:toFilter], clusterRoleBindings[toFilter+1:]...)
+	if oldestWithManagedByLabels != -1 {
+		return append(clusterRoleBindings[:oldestWithManagedByLabels], clusterRoleBindings[oldestWithManagedByLabels+1:]...)
+	}
+	return append(clusterRoleBindings[:oldestLegacy], clusterRoleBindings[oldestLegacy+1:]...)
```

Comment:
Here we're exposed to the issue described in 
- https://github.com/Kong/gateway-operator/issues/146